src/stepper.css.ts

```diff
@@ -0,0 +1,224 @@
+        backgroundColor: skinVars.colors.controlActivated,
+    },
+});
+
+const currentNumberTextKeyframes = keyframes({
+    '0%': {
+        color: skinVars.colors.textSecondary,
+    },
+    '100%': {
+        color: skinVars.colors.textPrimaryInverse,
+    },
+});
+
+const stepIconKeyframes = keyframes({
+    '0%': {
+        backgroundColor: skinVars.colors.controlActivated,
+        transform: 'scale(1)',
+    },
+    '100%': {
+        backgroundColor: skinVars.colors.controlActivated,
+        transform: 'scale(0)',
+    },
+});
+
+export const stepper = style([
+    sprinkles({
+        display: 'flex',
+        minHeight: 24,
+    }),
+    {
+        '@media': {
+            [mq.desktopOrBigger]: {
+                minHeight: stepperMinHeight,
+            },
+            [mq.tabletOrSmaller]: {
+                alignItems: 'center',
+            },
+        },
+    },
+]);
+
+export const step = style([
+    sprinkles({
+        position: 'relative',
+        display: 'inline-flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+    }),
+    {
+        textAlign: 'center',
+        ':first-child': {
+            alignItems: 'flex-start',
+            textAlign: 'left',
+        },
+        ':last-child': {
+            alignItems: 'flex-end',
+            textAlign: 'right',
+        },
+    },
+]);
+
+export const stepIconNumber = style([
+    sprinkles({
+        position: 'relative',
+        display: 'flex',
+    }),
+    {
+        height: `calc(${pxToRem(8)} + 16px)`,
+        width: `calc(${pxToRem(8)} + 16px)`,
+        '@media': {
+            [mq.desktopOrBigger]: {
+                height: `calc(${pxToRem(16)} + 16px)`,
+                width: `calc(${pxToRem(16)} + 16px)`,
+            },
+        },
+    },
+]);
+
+export const iconAnimation = style({
+    ':after': {
+        content: '""',
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        width: '100%',
+        height: '100%',
+        borderRadius: '50%',
+        animation: `${stepIconKeyframes} .3s ease-in-out`,
+        willChange: 'transform',
+        transition: 'transform .3s ease-in-out',
+    },
+});
+
+export const number = style([
+    sprinkles({
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        borderRadius: '50%',
+    }),
+    {
+        border: `2px solid ${skinVars.colors.borderDark}`,
+    },
+]);
+
+export const currentNumber = style({
+    transition: 'border-color .3s ease-in-out, background-color .3s ease-in-out',
+    willChange: 'border-color, background-color',
+    animation: `${currentNumberKeyframes} .3s ease-in-out`,
+    background: skinVars.colors.controlActivated,
+    borderColor: skinVars.colors.controlActivated,
+});
+
+globalStyle(`${currentNumber} span`, {
+    animation: `${currentNumberTextKeyframes} .3s ease-in-out`,
+    willChange: 'color',
+    transition: 'color .3s ease-in-out',
+});
```

Comment:
I think this could be the same animation (one than change border-color and color at once), and the global style removed

also changing children Text 
`isCurrent ? 'currentColor' : colors.textSecondary`

Reply:
I tried something like that but I wasn't getting the expected result